evalml/model_understanding/graphs.py

```diff
@@ -665,44 +732,50 @@ def graph_partial_dependence(pipeline, X, features, class_label=None, grid_resol
         fig = _subplots.make_subplots(rows=rows, cols=cols, subplot_titles=class_labels)
         for i, label in enumerate(class_labels):
             label_df = part_dep.loc[part_dep.class_label == label]
-            if mode == "two-way":
-                x = label_df.index
-                y = np.array([col for col in label_df.columns if isinstance(col, (int, float))])
-                z = label_df.values
-                fig.add_trace(_go.Contour(x=x, y=y, z=z, name=label, coloraxis="coloraxis"),
-                              row=(i + 2) // 2, col=(i % 2) + 1)
+            row = (i + 2) // 2
+            col = (i % 2) + 1
+            label_df.drop(columns=['class_label'], inplace=True)
+            if mode == 'two-way':
+                _update_fig_with_two_way_partial_dependence(_go, fig, label_df, part_dep, features, is_categorical,
+                                                            label, row, col)
             elif mode == "one-way":
                 x = label_df['feature_values']
                 y = label_df['partial_dependence']
-                fig.add_trace(_go.Scatter(x=x, y=y, line=dict(width=3), name=label),
-                              row=(i + 2) // 2, col=(i % 2) + 1)
+                if is_categorical:
+                    trace = _go.Bar(x=x, y=y, name=label)
+                else:
+                    trace = _go.Scatter(x=x, y=y, line=dict(width=3), name=label)
+                fig.add_trace(trace, row=(i + 2) // 2, col=(i % 2) + 1)
```

Comment:
we can simplify this to `row=row, col=col`, since you defined them inlines 735-736

Reply:
Good call! I missed this when I was moving everything around.